pkg/scheduler/apis/config/validation/validation.go

```diff
@@ -38,6 +38,10 @@ func ValidateKubeSchedulerConfiguration(cc *config.KubeSchedulerConfiguration) f
 	allErrs = append(allErrs, componentbasevalidation.ValidateLeaderElectionConfiguration(&cc.LeaderElection, field.NewPath("leaderElection"))...)
 
 	profilesPath := field.NewPath("profiles")
+	if cc.Parallelism <= 0 {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("parallelism"), cc.Parallelism, "should be an integer value greater than zero"))
+	}
```

Comment:
do we need a maximum? any other considerations we should requirement/document or require (power-of-two, etc)?

Reply:
Do we want to enforce them , or just add them to documentation as a reference ?

Reply:
I've never heard of non-power-of-two being a problem for parallelism. If anything, this should be up to the operator to decide if they want to use all the cores they have or not.

We could recommend, in the comment in types.go, to use up to the number of cores available in the system. But a global maximum doesn't necessarily make sense to me. We know current computers go up to about 256 cores, but I don't think we should hardcode that. On the other hand, a hard limit of 256 routines for a 2 core CPU wouldn't prevent an operator from achieving bad performance.